File: ios/TaskNotesApp/DeviceInfoModule.m
#import "DeviceInfoModule.h"
#import <UIKit/UIKit.h>
#import <sys/utsname.h>

@implementation DeviceInfoModule

RCT_EXPORT_MODULE(DeviceInfoBridge);

RCT_EXPORT_METHOD(getDeviceInfo:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
{
  @try {
    struct utsname systemInfo;
    uname(&systemInfo);

    NSString *deviceModel = [NSString stringWithCString:systemInfo.machine
                                              encoding:NSUTF8StringEncoding];

    NSDictionary *infoDictionary = [[NSBundle mainBundle] infoDictionary];
    NSString *appVersion = infoDictionary[@"CFBundleShortVersionString"] ?: @"1.0.0";
    NSString *buildNumber = infoDictionary[@"CFBundleVersion"] ?: @"1";

    NSDictionary *deviceInfo = @{
      @"deviceName": [[UIDevice currentDevice] name],
      @"systemVersion": [NSString stringWithFormat:@"iOS %@", [[UIDevice currentDevice] systemVersion]],
      @"appVersion": appVersion,
      @"buildNumber": buildNumber,
      @"platform": @"ios",
      @"model": deviceModel
    };

    resolve(deviceInfo);
  } @catch (NSException *exception) {
    reject(@"DEVICE_INFO_ERROR", @"Failed to get device info", nil);
  }
}

RCT_EXPORT_METHOD(getSystemVersion:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
{
  NSString *version = [NSString stringWithFormat:@"iOS %@",
                       [[UIDevice currentDevice] systemVersion]];
  resolve(version);
}

@end

File: android/app/src/main/java/com/tasknotesapp/deviceinfo/DeviceInfoModule.kt
package com.tasknotesapp.deviceinfo

import android.os.Build
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.WritableNativeMap

class DeviceInfoModule(reactContext: ReactApplicationContext) :
    ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String = "DeviceInfoBridge"

    @ReactMethod
    fun getDeviceInfo(promise: Promise) {
        try {
            val map = WritableNativeMap()
            map.putString("deviceName", "${Build.MANUFACTURER} ${Build.MODEL}")
            map.putString("systemVersion", "Android ${Build.VERSION.RELEASE}")
            map.putString("appVersion", "1.0.0")
            map.putString("buildNumber", "1")
            map.putString("platform", "android")
            promise.resolve(map)
        } catch (e: Exception) {
            promise.reject("DEVICE_INFO_ERROR", "Failed to get device info", e)
        }
    }

    @ReactMethod
    fun getSystemVersion(promise: Promise) {
        promise.resolve("Android ${Build.VERSION.RELEASE} (API ${Build.VERSION.SDK_INT})")
    }
}
